staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -106,9 +110,17 @@ func newETCD3HealthCheck(c storagebackend.Config) (func() error, error) {
 		// See https://github.com/etcd-io/etcd/blob/c57f8b3af865d1b531b979889c602ba14377420e/etcdctl/ctlv3/command/ep_command.go#L118
 		_, err := client.Get(ctx, path.Join("/", c.Prefix, "health"))
 		if err == nil {
-			return nil
+			healthzErrorMessage.Store("")
+			return
+		}
+		healthzErrorMessage.Store(err.Error())
+	}, time.Second, wait.NeverStop)
+
+	return func() error {
+		if errMsg := healthzErrorMessage.Load().(string); len(errMsg) > 0 {
+			return fmt.Errorf(errMsg)
```

Comment:
Use `fmt.Error`. Or consider just storing an error in the atomic value.

(I remember in the bad old days of C/C++ where this is a major bug!)

Reply:
Yeah, after looking at the rest of this, I think it'd be much better to store `error`s instead of `string`s in the atomic value.